app/code/Magento/Cms/Controller/Adminhtml/Page/PostDataProcessor.php

```diff
@@ -65,23 +68,25 @@ public function filter($data)
      */
     public function validate($data)
     {
-        $errorNo = true;
+        $hasError = false;
         if (!empty($data['layout_update_xml']) || !empty($data['custom_layout_update_xml'])) {
             /** @var $validatorCustomLayout \Magento\Framework\View\Model\Layout\Update\Validator */
             $validatorCustomLayout = $this->validatorFactory->create();
-            if (!empty($data['layout_update_xml']) && !$validatorCustomLayout->isValid($data['layout_update_xml'])) {
-                $errorNo = false;
-            }
-            if (!empty($data['custom_layout_update_xml'])
-                && !$validatorCustomLayout->isValid($data['custom_layout_update_xml'])
-            ) {
-                $errorNo = false;
-            }
-            foreach ($validatorCustomLayout->getMessages() as $message) {
-                $this->messageManager->addError($message);
+            try {
+                $hasError = $this->validateData($data, $validatorCustomLayout);
+            } catch (ValidationException $e) {
+            } catch (ValidationSchemaException $e) {
+            } catch (\Exception $e) {
+                $this->messageManager->addExceptionMessage($e);
+            } finally {
```

Comment:
I understand that you want to catch all other exceptions here but from a validation point I think there is only the `ValidationException` and `ValidationSchemaException` unless I am mistaken. In that case I would prefer the following.

```
try {
    $hasError = $this->validateData($data, $validatorCustomLayout);
} catch (ValidationException $e || ValidationSchemaException $e) {
    $validatorMessages = $validatorCustomLayout->getMessages();
    $hasError = $hasError || (bool) count($validatorMessages);
    foreach ($validatorMessages as $message) {
        $this->messageManager->addErrorMessage($message);
    }
} catch (\Exception $e) {
    $this->messageManager->addExceptionMessage($e);
}
```

What do you think?

Reply:
I hadn't seen that before, I mean, I didn't know I could catch several exceptions in a single catch block, I used finally to avoid logic duplication, but also that code must be executed if no exception is caught, so it can't be just there in the catch block. We have to find an intermediate solution.

Reply:
Is it really allowed to do that?
<img width="498" alt="captura de pantalla 2017-10-30 a las 12 38 37" src="https://user-images.githubusercontent.com/17545750/32169101-593d6ae8-bd6f-11e7-8af4-33f39e4be968.png">

If not, may I use instanceof a generic exception catch?